src/sema/eval.rs

```diff
@@ -196,3 +196,324 @@ pub fn eval_const_rational(
+
+            if let (
+                Some(Expression::NumberLiteral(_, _, left)),
+                Some(Expression::NumberLiteral(_, _, right)),
+            ) = (&left, &right)
+            {
+                Some(Expression::NumberLiteral(*loc, ty.clone(), left.sub(right)))
+            } else {
+                None
+            }
+        }
+
+        Expression::Multiply(loc, ty, _, left, right) => {
+            let left = eval_constants_in_expression(left, ns);
+            let right = eval_constants_in_expression(right, ns);
+
+            if let (
+                Some(Expression::NumberLiteral(_, _, left)),
+                Some(Expression::NumberLiteral(_, _, right)),
+            ) = (&left, &right)
+            {
+                Some(Expression::NumberLiteral(*loc, ty.clone(), left.mul(right)))
+            } else {
+                None
+            }
+        }
+        Expression::Divide(loc, ty, left, right) => {
+            let left = eval_constants_in_expression(left, ns);
+            let right = eval_constants_in_expression(right, ns);
+
+            if let (
+                Some(Expression::NumberLiteral(_, _, left)),
+                Some(Expression::NumberLiteral(_, _, right)),
+            ) = (&left, &right)
+            {
+                if right.is_zero() {
+                    ns.diagnostics
+                        .push(Diagnostic::error(*loc, "divide by zero".to_string()));
+                    None
+                } else {
+                    Some(Expression::NumberLiteral(*loc, ty.clone(), left.div(right)))
+                }
+            } else {
+                None
+            }
+        }
+
+        Expression::Modulo(loc, ty, left, right) => {
+            let left = eval_constants_in_expression(left, ns);
+            let right = eval_constants_in_expression(right, ns);
+
+            if let (
+                Some(Expression::NumberLiteral(_, _, left)),
+                Some(Expression::NumberLiteral(_, _, right)),
+            ) = (&left, &right)
+            {
+                if right.is_zero() {
+                    ns.diagnostics
+                        .push(Diagnostic::error(*loc, "divide by zero".to_string()));
+                    None
+                } else {
+                    Some(Expression::NumberLiteral(*loc, ty.clone(), left % right))
+                }
+            } else {
+                None
+            }
+        }
+        Expression::Power(loc, ty, _, left, right) => {
+            let left = eval_constants_in_expression(left, ns);
+            let right = eval_constants_in_expression(right, ns);
+
+            if let (
+                Some(Expression::NumberLiteral(_, _, left)),
+                Some(Expression::NumberLiteral(right_loc, _, right)),
+            ) = (&left, &right)
+            {
+                if let Some(diagnostic) = overflow_check(right, &Type::Uint(16), right_loc) {
+                    ns.diagnostics.push(diagnostic);
```

Comment:
This diagnostic is not right.

```solidity
uint x = 2 ** 14676683207225698178084221555689649093015162623576402558976;
```
And now I get an error about `uint16`, that is not clear. Please add same code as `ShiftLeft`